inc/global_content/namespace.php

```diff
@@ -0,0 +1,368 @@
+<?php
+/**
+ * Altis Global Content Repository.
+ *
+ * @package altis/core
+ */
+
+namespace Altis\Global_Content;
+
+use Exception;
+use WP_Admin_Bar;
+use WP_CLI;
+use WP_Site;
+
+/**
+ * Setup global media hooks.
+ *
+ * @return void
+ */
+function bootstrap() : void {
+	// Create the site on upgrade.
+	add_action( 'altis.migrate', __NAMESPACE__ . '\\maybe_create_site' );
```

Comment:
There should be another action triggered at `altis.migrate` that deals with setting a site that's been manually configured as the Global Content Site in the altis config. 

Something like (untested):

```php
add_action( 'altis.migrate', function() {
    $global_media_library_url = Altis\get_config()['media']['global-media-library'];

    // We're running this after the initial `altis.migrate` callback, so if we still don't have a global content site url at this stage, it means that we need to deal with a manually configured site.
    if ( ! get_site_url() && is_string( $global_media_library_url ) ) {
        $domain = stripslashes( parse_url( $global_media_library, PHP_URL_HOST ) );
        $path = stripslashes( parse_url( $global_media_library, PHP_URL_PATH ) );
        $site = get_site_by_path( $domain, $path );

        // Check for an error, e.g. if the site doesn't exist on the network and `get_site_by_path` couldn't find it.
        if ( ! WP_Error( $site ) ) {
   	    update_site_option( 'global_content_site_url', rtrim( $global_media_library_url, '/' ) );
	    update_site_option( 'global_content_site_id', $site->blog_id );
            update_site_meta( $site->blog_id, 'is_global_site', true );
            return;
        }
        // Do some error handling here. Possibly a try / catch would be better than an if / else.
    }
}, 11 );
```

This way the `Global_Content\get_site_id()` and similar functions will still work properly even with manually configured sites.

Reply:
This is assuming that a manually configured URL is part of the same network which is not the intention. It can be used to connect any other entirely separate WP site as the WP media source.

If the site is part of the same network we should control that process and set up in its entirety.

Reply:
The example code gives preference to sites on the network. If it wasn't on the network a `WP_Error` would be generated and nothing would be changed. In my eyes, if you _can_ manually set a URL to a site on the network, and you want to do that -- rather than using the site that is created automatically -- then the site shouldn't get treated like a second class citizen and not get any of the benefits/enhancements that you would otherwise with the global media site (not unless you don't want the additional restrictions). And if the only difference is setting a meta tag for a different site on the network, we should be able to handle that.

Reply:
Consider that you might not want to use the global content repo site as the media library, they don't have to be completely linked. I don't really get why you'd want the `global-media-library` setting from the media module to be able to modify which site is the Global Content Repo site. That's like the tail wagging the dog so to speak.

I added the option to set a URL for the `global-media-library` setting because it's a constant we can define that AMF WP works with so why restrict it?

This would make things more complicated IMO but we could update the docs for the media module to be more explicit about defining a URL being intended for external sites.

I get your intention regarding wanting to be able to define or use another site on the network more easily - so this would be a case of a project defining `AMF_WORDPRESS_URL` on an early hook. We can document how to do that at least but I don't think we should make any further assumptions about their intention for doing that in terms of applying the same modifications as we do to this global content site.

Reply:
this makes sense. I think what needs to be clarified then is the relationship (and distinction) between the Global _Content Site_ and the Global _Media Library_. The two terms are similar and, at least in this phase, are interchangeable, so the assumption is that defining the `global-media-library` in the `composer.json` would set the Global _Content Site_ (even though it doesn't strictly make sense since it's in the Media module, specifically).

Reply:
I'd say even now they aren't interchangeable, I've outlined in the docs in this repo that the Global Media Library is a feature built on top of the Global Content Repository.

I think that assumption might just be a problem for us in this team as that was the original plan, for anyone else we can present it in this new way from the outset. We can put it out to people and see if they make similar assumptions or ask questions along those lines later and make any changes accordingly.